CommonMark/Parser/BlockMethods.cs

```diff
@@ -92,16 +93,355 @@ private static void BreakOutOfLists(ref Block blockRef, LineInfo line)
 
-        public static void Finalize(Block b, LineInfo line)
+        static List<string> ParseTableLine(string line, StringBuilder sb)
+        {
+            var ret = new List<string>();
+
+            var i = 0;
+
+            if (i < line.Length && line[i] == '|') i++;
+
+            while (i < line.Length && char.IsWhiteSpace(line[i])) i++;
+
+            for (; i < line.Length; i++)
+            {
+                var c = line[i];
+                if (c == '\\')
+                {
+                    i++;
+                    continue;
+                }
+
+                if (c == '|')
+                {
+                    ret.Add(sb.ToString());
+                    sb.Clear();
+                }
+                else
+                {
+                    sb.Append(c);
+                }
+            }
+
+            if (sb.Length != 0)
+            {
+                ret.Add(sb.ToString());
+                sb.Clear();
+            }
+
+            return ret;
+        }
+
+        static void MakeTableCells(Block row, StringBuilder sb)
+        {
+            var asStr = row.StringContent.ToString();
+
+            var offset = 0;
+
+            for (var i = 0; i < asStr.Length; i++)
+            {
+                var c = asStr[i];
+
+                if (c == '|')
+                {
+                    var text = sb.ToString();
+                    sb.Clear();
+
+                    if (text.Length > 0)
+                    {
+                        var leadingWhiteSpace = 0;
+                        while (leadingWhiteSpace < text.Length && char.IsWhiteSpace(text[leadingWhiteSpace])) leadingWhiteSpace++;
+                        var trailingWhiteSpace = 0;
+                        while (trailingWhiteSpace < text.Length && char.IsWhiteSpace(text[text.Length - trailingWhiteSpace - 1])) trailingWhiteSpace++;
+
+                        var cell = new Block(BlockTag.TableCell, row.SourcePosition + offset + leadingWhiteSpace);
+                        cell.SourceLastPosition = cell.SourcePosition + text.Length - trailingWhiteSpace - leadingWhiteSpace;
+                        cell.StringContent = new StringContent();
+                        cell.StringContent.Append(text, leadingWhiteSpace, text.Length - leadingWhiteSpace - trailingWhiteSpace);
+
+                        if (row.LastChild == null)
+                        {
+                            row.FirstChild = row.LastChild = cell;
+                        }
+                        else
+                        {
+                            row.LastChild.NextSibling = cell;
+                            row.LastChild = cell;
+                        }
+
+                        cell.IsOpen = false;
+                    }
+
+                    offset += text.Length;
+
+                    // skip the |
+                    offset++;
+                    continue;
+                }
+
+                if (c == '\\')
+                {
+                    sb.Append(c);
```

Comment:
not all characters can be escaped in commonmark - that behavior should be used here as well: http://spec.commonmark.org/0.26/#backslash-escapes

see [InlineMethods.handle_backslash](https://github.com/Knagis/CommonMark.NET/blob/master/CommonMark/Parser/InlineMethods.cs#L591)